datahub-actions/src/datahub_actions/plugin/source/kafka/kafka_event_source.py

```diff
@@ -136,22 +141,68 @@ def __init__(self, config: KafkaEventSourceConfig, ctx: PipelineContext):
+            **async_commit_config,
+        }
+
+        if self.source_config.sasl_mechanism:
+            consumer_config["security.protocol"] = "SASL_SSL"
+            consumer_config["sasl.mechanism"] = self.source_config.sasl_mechanism
+
+            if self.source_config.sasl_mechanism == "OAUTHBEARER":
+                if not self.source_config.aws_region:
+                    raise ValueError(
+                        "aws_region must be set when using OAUTHBEARER (SASL/IAM)"
+                    )
+                try:
+                    # Delayed import to avoid making this a hard dependency if not used
+                    from aws_msk_iam_sasl_signer import MSKAuthTokenProvider
+                except ImportError as e:
+                    raise ImportError(
+                        "The 'aws_msk_iam_sasl_signer' library is required for SASL/IAM OAUTHBEARER. "
+                        "Please install it by adding 'aws-msk-iam-sasl-signer' to your project dependencies, "
+                        "or ensure your datahub installation includes kafka-iam extras."
+                    ) from e
+
+                def oauth_cb(oauth_config_str: str) -> tuple[str, float]:
+                    auth_token, expiry_ms = MSKAuthTokenProvider.generate_auth_token(
+                        self.source_config.aws_region  # type: ignore
+                    )
+                    return auth_token, expiry_ms / 1000
+
+                consumer_config["oauth_cb"] = oauth_cb
+            elif self.source_config.sasl_mechanism == "AWS_MSK_IAM":
+                # For librdkafka >= 1.5.0 with AWS_MSK_IAM mechanism,
+                # sasl.client.callback.handler.class and sasl.jaas.config
+                # are not needed and may cause errors if set, as librdkafka handles this internally.
+                # If "Unsupported SASL mechanism: AWS_MSK_IAM" occurs, it means this mechanism
+                # is not supported by the specific librdkafka build, despite its version.
+                # In that case, OAUTHBEARER with aws_msk_iam_sasl_signer is the fallback.
+                pass  # No extra config needed beyond sasl.mechanism itself
+            # Add other SASL mechanisms like SCRAM back here if general support is desired
+            # elif self.source_config.sasl_mechanism in ("SCRAM-SHA-256", "SCRAM-SHA-512"):
+            #    ... (add SCRAM logic if needed, including sasl_username/password from config)
+            else:
+                logger.warning(
+                    f"SASL mechanism '{self.source_config.sasl_mechanism}' is configured but not explicitly handled "
+                    f"by OAUTHBEARER or AWS_MSK_IAM specific logic. Ensure your librdkafka build supports it directly, "
+                    f"or that it doesn't require additional client-side configuration beyond 'sasl.mechanism'."
+                )
+
         self.consumer: confluent_kafka.Consumer = confluent_kafka.DeserializingConsumer(
```

Comment:
I think most of this should go into the `KafkaConsumerConnectionConfig` type - so that all places we connect to kafka can take advantage of sasl

Reply:
we do something similar with the SnowflakeConnectionConfig type - it embeds a bunch of logic so that you can just ask it to create the client for you

Reply:
@hsheth2 addressed